clarifai/client/input.py

```diff
@@ -0,0 +1,508 @@
+                - labels (list): A list of labels for the input.
+                - metadata (Struct): A Struct of metadata for the input.
+        Returns:
+            Input: An Input object for the specified input ID.
+        """
+        geo_pb = resources_pb2.Geo(geo_point=resources_pb2.GeoPoint(
+            longitude=kwargs['geo_info'][0], latitude=kwargs['geo_info'][
+                1])) if 'geo_info' in kwargs else None
+        concepts=[
+                resources_pb2.Concept(
+                id=f"id-{''.join(_label.split(' '))}", name=_label, value=1.)\
+                for _label in kwargs['labels']
+            ]if 'labels' in kwargs else None
+        metadata = kwargs['metadata'] if 'metadata' in kwargs else None
+
+        if dataset_id:
+            return resources_pb2.Input(
+                id=input_id,
+                dataset_ids=[dataset_id],
+                data=resources_pb2.Data(image=imagepb, video=video_pb, audio=audio_pb, text=text_pb, geo=geo_pb, concepts=concepts, metadata=metadata))
+
+        return resources_pb2.Input(
+            id=input_id,
+            data=resources_pb2.Data(image=imagepb, video=video_pb, audio=audio_pb, text=text_pb, geo=geo_pb, concepts=concepts, metadata=metadata))
+
+    def get_input_from_url(self, input_id: str , image_url: str = None, video_url: str = None, audio_url: str = None, text_url:str = None ,dataset_id: str = None, **kwargs) -> Input:
+        """Create input proto from url.
+        Args:
+            input_id (str): The input ID for the input to create.
+            image_url (str): The url for the image.
+            video_url (str): The url for the video.
+            audio_url (str): The url for the audio.
+            text_url (str): The url for the text.
+                dataset_id (str): The dataset ID for the dataset to add the input to.
+        Returns:
+            Input: An Input object for the specified input ID.
+        Example:
+            >>> from clarifai.client.input import Input
+            >>> input_obj = Input()
+            >>> input_proto = input_obj.get_input_from_url(input_id = 'demo', image_url='https://samples.clarifai.com/metro-north.jpg')
+        """
+        if not any((image_url, video_url, audio_url, text_url)):
+            raise ValueError("At least one of image_url, video_url, audio_url, text_url must be provided.")
+        image_pb = resources_pb2.Image(url=image_url) if image_url else None
+        video_pb = resources_pb2.Video(url=video_url) if video_url else None
+        audio_pb = resources_pb2.Audio(url=audio_url) if audio_url else None
+        text_pb = resources_pb2.Text(url=text_url) if text_url else None
+        return self._get_proto(input_id, dataset_id, image_pb, video_pb,audio_pb,text_pb, **kwargs)
+
+    def get_input_from_filename(self, input_id: str , image_file: str = None, video_file: str = None, audio_file: str = None, dataset_id: str = None, **kwargs) -> Input:
```

Comment:
Should it just be `get_input_from_file`?

Reply:
Yeah, `get_input_from_file` looks better!! 👍